owltorrent-network/src/main/java/edu/rice/owltorrent/network/MeteringSystem.java

```diff
@@ -0,0 +1,194 @@
+package edu.rice.owltorrent.network;
+
+import com.google.common.util.concurrent.AtomicDouble;
+import edu.rice.owltorrent.common.entity.TwentyByteId;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.PriorityQueue;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Metering system to keep track of peer, system, and piece-related metrics.
+ *
+ * @author bhaveshshah
+ */
+public class MeteringSystem {
+  public enum Metrics {
+    // Peer-related metrics
+    PEER_DOWNLOAD_TIME,
+
+    // System-related metrics
+    ENTIRE_DOWNLOAD_TIME,
+    EFFECTIVE_PIECES,
+    FAILED_PIECES,
+
+    // Piece-related metrics
+    PIECE_DOWNLOAD_TIME
+  }
+
+  // Maps to store metrics and associated values
+  public static ConcurrentHashMap<TwentyByteId, ConcurrentHashMap<Enum<?>, AtomicDouble>>
```

Comment:
Why are these static?

Reply:
I'm not sure, but feel like it should be private final instead?

Reply:
Changed in next commit